cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/ArtifactSelectorProvider.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.common;
+
+import co.cask.cdap.api.artifact.ArtifactScope;
+import co.cask.cdap.api.artifact.ArtifactVersion;
+import co.cask.cdap.api.plugin.PluginSelector;
+import co.cask.cdap.etl.proto.ArtifactSelectorConfig;
+import co.cask.cdap.proto.artifact.ArtifactVersionRange;
+import co.cask.cdap.proto.artifact.InvalidArtifactRangeException;
+import com.google.common.base.CharMatcher;
+
+import javax.annotation.Nullable;
+
+/**
+ * Provides the {@link ArtifactSelector} from a {@link ArtifactSelectorConfig}.
+ */
+public class ArtifactSelectorProvider {
+  private static final CharMatcher nameMatcher =
+    CharMatcher.inRange('A', 'Z')
+      .or(CharMatcher.inRange('a', 'z'))
+      .or(CharMatcher.inRange('0', '9'))
+      .or(CharMatcher.is('_'))
+      .or(CharMatcher.is('-'));
+
+  private final ArtifactSelectorConfig config;
+  private final String pluginType;
+  private final String pluginName;
+
+  public ArtifactSelectorProvider(@Nullable ArtifactSelectorConfig config, String pluginType, String pluginName) {
```

Comment:
it would be more appropriate for this the constructor to only take the plugin type and name, and for getPluginSelector() to take the ArtifactSelectorConfig. Otherwise this isn't really a provider, as it can only be used once.